lib/dictBuilder/cover.c

```diff
@@ -929,6 +929,111 @@ void COVER_best_finish(COVER_best_t *best, size_t compressedSize,
   }
 }
 
+COVER_dictSelection_t COVER_dictSelectionError(size_t error) {
+    const COVER_dictSelection_t selection = { NULL, 0, error };
+    return selection;
+}
+
+unsigned COVER_dictSelectionIsError(COVER_dictSelection_t selection) {
+  return (ZSTD_isError(selection.totalCompressedSize) || !selection.dictContent);
+}
+
+void COVER_dictSelectionFree(COVER_dictSelection_t selection){
+  free(selection.dictContent);
+}
+
+COVER_dictSelection_t COVER_selectDict(BYTE* customDictContent,
+        size_t dictContentSize, const BYTE* samplesBuffer, const size_t* samplesSizes, unsigned nbFinalizeSamples,
+        size_t nbCheckSamples, size_t nbSamples, ZDICT_cover_params_t params, size_t* offsets, size_t totalCompressedSize) {
+
+  size_t largestDict = 0;
+  size_t largestCompressed = 0;
+  BYTE* customDictContentEnd = customDictContent + dictContentSize;
+
+  BYTE *const largestDictbuffer = (BYTE * const)malloc(dictContentSize);
+  BYTE *const candidateDictBuffer = (BYTE * const)malloc(dictContentSize);
+  double regressionTolerance = params.shrinkDict + 1.00;
```

Comment:
I liked taking the parameter as an `unsigned`, but we should distinguish:

1. Auto shrinking enabled vs disabled. If you need fast dictionary training, you don't want auto shrinking.
   It can be disabled by default.
2. The regression we allow. A regression of 0 could make sense if you want to take a smaller dictionary only if it is better.

I think we could take two flags `shrinkDict` and `shrinkDictMaxRegression`, both `unsigned`. It was a mistake to take `split` as a double, so I don't want to repeat it here.

On the library side, we can default to disabled, and the regression defaults to 0.

On the CLI, we can default to disabled, but we could say if someone sets `--train-fastcover=shrink` we enable with default to a regression of 1, but `--train-fastcover=shrink=0` or `--train-fastcover=shrink=2` should both still work.